src/Engines/engine.ts

```diff
@@ -994,7 +994,22 @@ export class Engine extends ThinEngine {
      * @param depthFunc defines the function to use
      */
     public setDepthFunction(depthFunc: number) {
-        this._depthCullingState.depthFunc = depthFunc;
+        // We delegate these calls to the named functions because
+        // NativeEngine overrides them to translate the constants.
+        switch (depthFunc) {

```

Comment:
We would in this case miss quite a few of them: 
gl.NEVER (never pass)
gl.LESS (pass if the incoming value is less than the depth buffer value)
gl.EQUAL (pass if the incoming value equals the depth buffer value)
gl.LEQUAL (pass if the incoming value is less than or equal to the depth buffer value)
gl.GREATER (pass if the incoming value is greater than the depth buffer value)
gl.NOTEQUAL (pass if the incoming value is not equal to the depth buffer value)
gl.GEQUAL (pass if the incoming value is greater than or equal to the depth buffer value)
gl.ALWAYS (always pass)

Reply:
This is used with more cases than the few helper we have on the Engine class. I guess all the helpers should call into this one which is more generic and native could only implement this one instead ?

Reply:
Whoops, completely forgot about the other cases.

Reply:
nw I almost did to :-)